modules/javafx.graphics/src/jslc/java/com/sun/scenario/effect/compiler/tree/JSLVisitor.java

```diff
@@ -0,0 +1,756 @@
+            return tm.fieldSelect(visitPrimary_or_call(ctx.e), visitField_selection(ctx.fs).getString());
+        } else if (ctx.INC() != null) {
+            return tm.unary(UnaryOpType.INC, visitPrimary_or_call(ctx.e));
+        } else if (ctx.DEC() != null) {
+            return tm.unary(UnaryOpType.DEC, visitPrimary_or_call(ctx.e));
+        } else if (ctx.e != null){
+            return visitPrimary_or_call(ctx.e);
+        }
+
+        throw new RuntimeException("invalid postfix expression");
+    }
+
+    @Override
+    public Expr visitFunction_call(JSLParser.Function_callContext ctx) {
+        if (ctx.IDENTIFIER() != null) {
+            return tm.call(ctx.IDENTIFIER().getText(), ctx.function_call_parameter_list() != null ?
+                    visitFunction_call_parameter_list(ctx.p).getParams() : null);
+        } else if (ctx.type_specifier() != null) {
+            Type type = Type.fromToken(ctx.ts.getText());
+            return tm.vectorCtor(type, ctx.function_call_parameter_list() != null ?
+                    visitFunction_call_parameter_list(ctx.p).getParams() : null);
+        }
+
+        throw new RuntimeException("invalid function call");
+    }
+
+    @Override
+    public CallExpr visitFunction_call_parameter_list(JSLParser.Function_call_parameter_listContext ctx) {
+        List<Expr> list = new ArrayList<>();
+        for (JSLParser.Assignment_expressionContext aeCtx : ctx.assignment_expression()) {
+            list.add(visitAssignment_expression(aeCtx));
+        }
+        return new CallExpr(null, list);
+    }
+
+    @Override
+    public Expr visitUnary_expression(JSLParser.Unary_expressionContext ctx) {
+        if (ctx.INC() != null) {
+            return tm.unary(UnaryOpType.INC, visitUnary_expression(ctx.u));
+        } else if (ctx.DEC() != null) {
+            return tm.unary(UnaryOpType.DEC, visitUnary_expression(ctx.u));
+        } else if (ctx.PLUS() != null) {
+            return tm.unary(UnaryOpType.PLUS, visitUnary_expression(ctx.u));
+        } else if (ctx.DASH() != null) {
+            return tm.unary(UnaryOpType.MINUS, visitUnary_expression(ctx.u));
+        } else if (ctx.BANG() != null) {
+            return tm.unary(UnaryOpType.NOT, visitUnary_expression(ctx.u));
+        } else if (ctx.postfix_expression() != null){
+            return visitPostfix_expression(ctx.p);
+        }
+
+        throw new RuntimeException("invalid unary expression");
+    }
+
+    @Override
+    public Expr visitMultiplicative_expression(JSLParser.Multiplicative_expressionContext ctx) {
+        Expr expr = visitUnary_expression(ctx.a);
+
+        if (ctx.multiplicative_operator() != null) {
+            for (int i = 0; i < ctx.multiplicative_expression().size(); i++) {
+                JSLParser.Multiplicative_expressionContext context = ctx.multiplicative_expression(i);
+                switch (ctx.multiplicative_operator(i).getText()) {
+                    case "*":
```

Comment:
Would it be possible to use either an enum or a `TerminalNode` rather than checking against the string representation of the operator? If so, that might be cleaner -- it seems better to use `STAR()` than `"*"` here, similar to what you did for the unary operators.